datastream-jdbc-connector/src/main/java/com/linkedin/datastream/connectors/jdbc/translator/JdbcCommon.java

```diff
@@ -335,9 +331,14 @@ public static long convertToAvroStream(final ResultSet rs, final OutputStream ou
                             // The cause of the exception is not known, but we'll fall back to call getObject() and handle any "real" exception there
                             value = rs.getObject(i);
                         }
-                    } else if (javaSqlType == microsoft.sql.Types.DATETIMEOFFSET) { // -155 represents TSQL Datetimeoffset type
+                    } else if (javaSqlType == TIMESTAMP // 93
+                            || javaSqlType == TIMESTAMP_WITH_TIMEZONE // 2014
+                            || javaSqlType == microsoft.sql.Types.SMALLDATETIME // -150 represents TSQL smalldatetime type
+                            || javaSqlType == microsoft.sql.Types.DATETIME // -151 represents TSQL datetime type
+                            || javaSqlType == microsoft.sql.Types.DATETIMEOFFSET) { // -155 represents TSQL Datetimeoffset type
+
                         try {
-                            // Convert datetimeoffset to UTC then convert to timestamp millis
+                            // Convert datetime, datetime2 and datetimeoffset to UTC then convert to timestamp millis
```

Comment:
Good approach. But, you need pass additional info through schema.

The problem is, when we convert all these types to logical type timestamp-millis, we lose the information on which sql field type the value is coming from. 

As you know, you will not be able to properly convert non datetimeoffset type to UTC value. So you leave it to the consumer to infer the timezone. The consumer can only infer the timezone properly when it knows the type of the source field.

You don't have to change anything here. See my comment on schema.